crates/weaver_version/src/schema_changes.rs

```diff
@@ -0,0 +1,307 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Data structures and utilities for tracking schema changes between versions.
+
+use serde::Serialize;
+use std::collections::HashMap;
+
+/// The type of schema item.
+#[derive(Debug, Serialize, Hash, Eq, PartialEq, Copy, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum SchemaItemType {
+    /// Registry attributes
+    RegistryAttributes,
+    /// Metrics
+    Metrics,
+    /// Events
+    Events,
+    /// Spans
+    Spans,
+    /// Resources
+    Resources,
+}
+
+/// A summary of schema changes between two versions of a schema.
+#[derive(Debug, Default, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub struct SchemaChanges {
+    /// Information on the registry manifest for the most recent version of the schema.
+    head: RegistryManifest,
+
+    /// Information of the registry manifest for the baseline version of the schema.
+    baseline: RegistryManifest,
+
+    /// A map where the key is the type of schema item (e.g., "attributes", "metrics",
+    /// "events, "spans", "resources"), and the value is a list of changes associated
+    /// with that item type.
+    changes: HashMap<SchemaItemType, Vec<SchemaItemChange>>,
+}
+
+/// Represents the information of a semantic convention registry manifest.
+#[derive(Debug, Default, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub struct RegistryManifest {
+    /// The version of the registry which will be used to define the semconv package version.
+    pub semconv_version: String,
+}
+
+/// Represents the different types of changes that can occur between
+/// two versions of a schema. This covers changes such as adding, removing,
+/// renaming, and deprecating telemetry objects (attributes, metrics, etc.).
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "snake_case")]
+#[serde(tag = "type")]
+pub enum SchemaItemChange {
+    /// A top-level telemetry object (e.g., attribute, metric, etc.) was added to the head registry.
+    Added {
+        /// The name of the added telemetry object.
+        name: String,
+    },
+    /// A top-level telemetry object from the baseline registry was renamed in the head registry.
+    Renamed {
+        /// The old name of the telemetry object that has been renamed.
+        old_name: String,
+        /// The new name of the telemetry object that has been renamed.
+        new_name: String,
+        /// A note providing further context.
+        note: String,
+    },
+    /// A top-level telemetry object from the baseline registry was marked as deprecated in the head
+    /// registry.
+    Deprecated {
```

Comment:
Is this its own change or should it be attached to other changes?

I.e. is it just an implication of change?  

I think this was called out verbally, but it's the one I'm least sure of belonging with other "semantic" changes, especialyl given "uncategorized" as an option.

Reply:
See my comment below.

Reply:
Responded.  Once updated, this PR LGTM.

Structurally/rust-wise you have all the pieces I'd look for.  It's just naming/surface syntax at this point.